### PR TITLE
State by what component a system is not supported in a generic message

### PR DESCRIPTION
**Pull request by yarikoptic**
Otherwise it is hard to impossible (line numbers no longer match, pain) to determine who is complaining and why in a complex invocation composition

### CODE REVIEW

File: src/datalad_installer.py [hunk 1]
```diff
@@ -2084,7 +2084,7 @@ def install_package(
 
     def assert_supported_system(self, **kwargs: Any) -> None:
         if not (ON_LINUX or ON_MACOS or ON_WINDOWS):
-            raise MethodNotSupportedError(f"{SYSTEM} OS not supported")
+            raise MethodNotSupportedError(f"{SYSTEM} OS not supported by {self}")
```
Comment (jwodder) [inline src/datalad_installer.py:2087]: I think these should be `{self.NAME}` instead, as that shows the name used on the command line.

**Review by jwodder — CHANGES_REQUESTED** [review]

File: src/datalad_installer.py (hunk 1 above)
Comment (yarikoptic) [reply src/datalad_installer.py:2087]: ok, let's do that

```suggestion
            raise MethodNotSupportedError(f"{SYSTEM} OS not supported by {self.NAME}")
```
?

while at it -- is there a way to include NAME of the component component (git-annex, etc) in addition?

Comment (jwodder) [reply src/datalad_installer.py:2087]: You would have to pass `component` to `assert_supported_system()` on line 1468 and update all the `assert_supported_system()` definitions to include the string in the error message.

Comment (yarikoptic) [reply src/datalad_installer.py:2087]: ok, let's then skip that for now  -- too much for `sed` ;)

**Review by jwodder — APPROVED** [review]

**Pull request merged**